### PR TITLE
Refactor and reorganize the 3.0 docs in the Deploy section

### PR DESCRIPTION
**Pull request by djsauble**
This PR is an attempt to refactor and reorganize all of the 3.0 deploy docs to minimize the amount of overlap between docs and to make the relationships between pages clearer at a glance.

* Establish two main categories of infra docs: concepts and examples.
* Create an umbrella **Deployments overview** page with an overview of what deployments are and why they're useful.
* Merge content that is overly similar so that there's a single source of truth for major topics (e.g. how to use `deploy()` with Docker work pools).
* Make page titles use the same consistent "&lt;verb&gt; &lt;noun&gt;" format that's used elsewhere.

Closes https://github.com/PrefectHQ/prefect/issues/14094

### Example

Before:

image

After:

image

### Checklist

- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
- [x] This pull request references any related issue by including "closes `<link to issue>`"
  
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: docs/3.0rc/deploy/index.mdx [hunk 1]
```diff
@@ -12,7 +12,9 @@ search:
   boost: 2
```
Comment (discdiver) [inline docs/3.0rc/deploy/index.mdx:12]:
```suggestion
```

File: docs/3.0rc/deploy/index.mdx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ---
-title: Deploy flows
-description: Learn how to configure flows for scheduled and remote execution.
+title: Deployment overview
+description: Learn what deployments are and how to use them in Prefect.
 tags:
```
Comment (discdiver) [inline docs/3.0rc/deploy/index.mdx:4]:
```suggestion
```

Comment (discdiver) [reply docs/3.0rc/deploy/index.mdx:4]: tags and search->boost are left over metadata from mkdocs that we should remove.

File: docs/3.0rc/deploy/index.mdx [hunk 3]
```diff
@@ -12,7 +12,9 @@ search:
   boost: 2
 ---
 
-[Deployments](/3.0rc/deploy/serve-flows/) are server-side representations of flows. 
+## What is a deployment?
```
Comment (discdiver) [inline docs/3.0rc/deploy/index.mdx:15]:
```suggestion
## Deployments
```

Comment (discdiver) [reply docs/3.0rc/deploy/index.mdx:15]: Stellar was in favor of not leading with a question. Probably good to stick with that for consistency. We should add in our style guide.

File: docs/3.0rc/deploy/index.mdx [hunk 4]
```diff
@@ -218,7 +72,7 @@ class Deployment:
 All methods for creating Prefect deployments are interfaces for populating this schema. 
 The following sections explain each component in more detail.
 
-### Required data
+### Required defining data
 
 Deployments universally require a `name` and a reference to an underlying `Flow`.
```
Comment (discdiver) [inline docs/3.0rc/deploy/index.mdx:77]:
```suggestion
Deployments require a `name` and a reference to an underlying `Flow`.
```

File: docs/3.0rc/deploy/index.mdx [hunk 5]
```diff
@@ -289,73 +143,7 @@ passed to a flow run against the schema defined by `parameter_openapi_schema`.
...
-**Using the Prefect client**
-
-For more information on using the Prefect client to interact with Prefect's
-REST API, see [Interact with the API](/3.0rc/manage/interact-with-api/).
-</Tip>
-
-## Versioning and bookkeeping
+### Metadata for bookkeeping
 
 Here is important information on the versions, descriptions, and tags fields:
```
Comment (discdiver) [inline docs/3.0rc/deploy/index.mdx:148]:
```suggestion
Important information for the versions, descriptions, and tags fields:
```

File: docs/3.0rc/deploy/index.mdx [hunk 6]
```diff
@@ -422,23 +215,22 @@ run is a suitable deployment environment.
 - Creating deployments requires a minimal set of decisions.
 - Iteration speed is fast.
 
-### Dynamically provisioning infrastructure with work pools
+### Dynamic infrastructure
 
-Consider running flows on dynamically provisioned 
-infrastructure with work pools under these circumstances:
+Consider running flows on dynamically provisioned infrastructure with work pools under these circumstances:
```
Comment (discdiver) [inline docs/3.0rc/deploy/index.mdx:220]:
```suggestion
Consider running flows on dynamically provisioned infrastructure with work pools when you have any of the following:
```

File: docs/3.0rc/deploy/index.mdx [hunk 7]
```diff
@@ -422,23 +215,22 @@ run is a suitable deployment environment.
...
 
 - Flows that have expensive infrastructure due to the long-running process.
```
Comment (discdiver) [inline docs/3.0rc/deploy/index.mdx:222]:
```suggestion
- Flows that requier expensive infrastructure due to the long-running process.
```

File: docs/3.0rc/deploy/index.mdx [hunk 8]
```diff
@@ -422,23 +215,22 @@ run is a suitable deployment environment.
...
 - Flows with heterogeneous infrastructure needs across runs.
 - Large volumes of deployments.
 - If your internal team structure requires that deployment authors be members of a different team
```
Comment (discdiver) [inline docs/3.0rc/deploy/index.mdx:225]:
```suggestion
- An internal organizational structure in which deployment authors or runners are not members of
```

File: docs/3.0rc/deploy/index.mdx [hunk 9]
```diff
@@ -422,23 +215,22 @@ run is a suitable deployment environment.
...
 
-Consider running flows on dynamically provisioned 
-infrastructure with work pools under these circumstances:
+Consider running flows on dynamically provisioned infrastructure with work pools under these circumstances:
 
 - Flows that have expensive infrastructure due to the long-running process.
 - Flows with heterogeneous infrastructure needs across runs.
 - Large volumes of deployments.
 - If your internal team structure requires that deployment authors be members of a different team 
 than the team managing infrastructure.
```
Comment (discdiver) [inline docs/3.0rc/deploy/index.mdx:226]:
```suggestion
the team that manages the infrastructure.
```

File: docs/3.0rc/deploy/infrastructure-examples/managed.mdx [hunk 10]
```diff
@@ -119,8 +106,8 @@ You can view your compute hours quota usage on the **Work Pools** page in the UI
 
 ## Next steps
 
-Read more about creating deployments in the [deployment guide](/3.0rc/deploy/control-runs/prefect-deploy/).
+Read more about creating deployments in the [deployment guide](/3.0rc/deploy/infrastructure-examples/docker/).
```
Comment (discdiver) [inline docs/3.0rc/deploy/infrastructure-examples/managed.mdx:109]: We probably don't want to call this the deployment guide.

File: docs/3.0rc/deploy/infrastructure-examples/serverless.mdx [hunk 11]
```diff
@@ -1,9 +1,9 @@
 ---
-title: Push work to serverless computing infrastructure
+title: Run flows on serverless infra
```
Comment (discdiver) [inline docs/3.0rc/deploy/infrastructure-examples/serverless.mdx:2]:
```suggestion
title: Run flows on serverless infrastructure
```

I think our unwritten style guide has us spelling out infrastructure. We should make that explicit. :)

File: docs/3.0rc/deploy/infrastructure-examples/serverless.mdx [hunk 12]
```diff
@@ -629,14 +629,14 @@ Click **Create** to configure your push work pool by selecting a push option in
  <Tab title="Google Cloud Run">
 
     Each step has several optional fields that are detailed in the 
-    [work pools guide](/3.0rc/deploy/control-runs/work-pools/). 
+    [work pools guide](/3.0rc/deploy/infrastructure-concepts/work-pools/).
```
Comment (discdiver) [inline docs/3.0rc/deploy/infrastructure-examples/serverless.mdx:632]:
```suggestion
    [work pools docs](/3.0rc/deploy/infrastructure-concepts/work-pools/). 
```

**Review by discdiver — COMMENTED** [review]
Nice! Minor suggestions.

File: docs/3.0rc/deploy/index.mdx [hunk 13]
```diff
@@ -1,6 +1,6 @@
 ---
-title: Deploy flows
-description: Learn how to configure flows for scheduled and remote execution.
+title: Deployment overview
+description: Learn what deployments are and how to use them in Prefect.
```
Comment (billpalombi) [inline docs/3.0rc/deploy/index.mdx:3]:
```suggestion
description: Learn how to use deployments to trigger flow runs remotely.
```

File: docs/3.0rc/deploy/infrastructure-examples/serverless.mdx (hunk 11 above)
Comment (billpalombi) [reply docs/3.0rc/deploy/infrastructure-examples/serverless.mdx:2]: If infrastructure makes the title wrap, consider "compute" instead

File: docs/3.0rc/deploy/infrastructure-examples/serverless.mdx (hunk 12 above)
Comment (billpalombi) [reply docs/3.0rc/deploy/infrastructure-examples/serverless.mdx:632]: I would use "page" instead.

File: docs/integrations/prefect-docker/index.mdx [hunk 14]
```diff
@@ -31,7 +31,7 @@ Install or update to the latest version of the prefect-docker library and depend
 </Tabs>
 
 ### Examples
-See the Prefect [Workers docs](3.0rc/deploy/dynamic-infra/push-runs-remote) to learn how to create and run deployments that use Docker.
+See the Prefect [Workers docs](/3.0rc/deploy/static-infra/push-runs-remote) to learn how to create and run deployments that use Docker.
```
Comment (billpalombi) [inline docs/integrations/prefect-docker/index.mdx:34]: I think this link is no longer accurate

File: docs/3.0rc/deploy/index.mdx (hunk 7 above)
Comment (daniel-prefect) [reply docs/3.0rc/deploy/index.mdx:222]:
```suggestion
- Flows that require expensive infrastructure due to the long-running process.
```

File: docs/3.0rc/deploy/infrastructure-examples/managed.mdx (hunk 10 above)
Comment (daniel-prefect) [reply docs/3.0rc/deploy/infrastructure-examples/managed.mdx:109]:
```suggestion
Read more about creating deployments in [Run flows in Docker containers](/3.0rc/deploy/infrastructure-examples/docker/).
```

File: docs/3.0rc/deploy/infrastructure-examples/serverless.mdx (hunk 11 above)
Comment (daniel-prefect) [reply docs/3.0rc/deploy/infrastructure-examples/serverless.mdx:2]:
```suggestion
title: Run flows on serverless compute
```

File: docs/3.0rc/deploy/infrastructure-examples/serverless.mdx (hunk 12 above)
Comment (daniel-prefect) [reply docs/3.0rc/deploy/infrastructure-examples/serverless.mdx:632]: The Google Developer Docs style guide suggests making the link text match the exact text of the title or heading you're referencing. https://developers.google.com/style/link-text#structure-link-text

```suggestion
    [Manage infrastructure with work pools](/3.0rc/deploy/infrastructure-concepts/work-pools/). 
```

File: docs/3.0rc/deploy/infrastructure-examples/serverless.mdx [hunk 15]
```diff
@@ -629,14 +629,14 @@ Click **Create** to configure your push work pool by selecting a push option in
  <Tab title="Google Cloud Run">
 
     Each step has several optional fields that are detailed in the
```
Comment (daniel-prefect) [inline docs/3.0rc/deploy/infrastructure-examples/serverless.mdx:631]:
```suggestion
    Each step has several optional fields that are detailed in
```

File: docs/integrations/prefect-docker/index.mdx (hunk 14 above)
Comment (daniel-prefect) [reply docs/integrations/prefect-docker/index.mdx:34]: Good catch, I missed one. 😅

**Review by discdiver — APPROVED** [review]

**Pull request merged**